lib/CMakeLists.txt

```diff
@@ -64,27 +70,51 @@ if (SWIFT_SWIFT_PARSER)
       )
     endif()
 
-    add_custom_command(
-      OUTPUT "${SWIFT_HOST_LIBRARIES_DEST_DIR}/${sharedlib}"
-      DEPENDS "${SWIFT_SYNTAX_LIBRARIES_BUILD_DIR}/${sharedlib}"
-      COMMAND ${CMAKE_COMMAND} -E copy_if_different ${SWIFT_SYNTAX_LIBRARIES_BUILD_DIR}/${sharedlib} ${SWIFT_HOST_LIBRARIES_DEST_DIR}/${sharedlib}
-      ${add_origin_rpath}
-    )
-
-    add_custom_target(copy_swiftSyntaxLibrary_${sharedlib}
-      DEPENDS "${SWIFT_HOST_LIBRARIES_DEST_DIR}/${sharedlib}"
-      COMMENT "Copying ${sharedlib}"
-    )
-
-    swift_install_in_component(
-      PROGRAMS "${SWIFT_HOST_LIBRARIES_DEST_DIR}/${sharedlib}"
-      DESTINATION "lib${LLVM_LIBDIR_SUFFIX}/swift/host"
-      COMPONENT compiler
-    )
+    if(CMAKE_SYSTEM_NAME MATCHES Windows)
+      add_custom_command(OUTPUT ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${sharedlib}
+                         DEPENDS "${SWIFT_PATH_TO_EARLYSWIFTSYNTAX_BUILD_DIR}/bin/${sharedlib}"
+                         COMMAND ${CMAKE_COMMAND} -E copy_if_different ${SWIFT_PATH_TO_EARLYSWIFTSYNTAX_BUILD_DIR}/bin/${sharedlib} ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${sharedlib})
+      add_custom_target(copy_swiftSyntaxLibrary_${sharedlib}
+                        DEPENDS ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${sharedlib}
+                        COMMENT "copying ${sharedlib}")
+      swift_install_in_component(PROGRAMS ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${sharedlib}
+                                 DESTINATION bin
+                                 COMPONENT compiler)
+    else()
+      add_custom_command(OUTPUT "${SWIFT_HOST_LIBRARIES_DEST_DIR}/${sharedlib}"
+                         DEPENDS "${SWIFT_SYNTAX_LIBRARIES_BUILD_DIR}/${sharedlib}"
+                         COMMAND ${CMAKE_COMMAND} -E copy_if_different ${SWIFT_SYNTAX_LIBRARIES_BUILD_DIR}/${sharedlib} ${SWIFT_HOST_LIBRARIES_DEST_DIR}/${sharedlib}
+                         ${add_origin_rpath})
+      add_custom_target(copy_swiftSyntaxLibrary_${sharedlib}
+                        DEPENDS "${SWIFT_HOST_LIBRARIES_DEST_DIR}/${sharedlib}"
+                        COMMENT "Copying ${sharedlib}")
+      swift_install_in_component(PROGRAMS "${SWIFT_HOST_LIBRARIES_DEST_DIR}/${sharedlib}"
+                                 DESTINATION "lib${LLVM_LIBDIR_SUFFIX}/swift/host"
+                                 COMPONENT compiler)
+    endif()
 
     add_dependencies(swiftSyntaxLibraries copy_swiftSyntaxLibrary_${sharedlib})
   endforeach()
 
+  foreach(implib ${SWIFT_SYNTAX_IMPORT_LIBRARIES})
```

Comment:
Needs a Windows block

Reply:
Hmm, it should be an empty for loop for other targets, why would this need a Windows or AIX block?

Reply:
Why is it empty? It's defined by:
```
 list(TRANSFORM SWIFT_SYNTAX_MODULES PREPEND "${CMAKE_IMPORT_LIBRARY_PREFIX}"
    OUTPUT_VARIABLE SWIFT_SYNTAX_IMPORT_LIBRARIES)
  list(TRANSFORM SWIFT_SYNTAX_IMPORT_LIBRARIES APPEND
    "${CMAKE_IMPORT_LIBRARY_SUFFIX}" OUTPUT_VARIABLE
    SWIFT_SYNTAX_IMPORT_LIBRARIES)
```

Wouldn't it just be the original `SWIFT_SYNTAX_MODULES` list (since both the prefix and suffix will be empty)?

Reply:
I thought it was used in a glob somewhere :/